### ISSUE TITLE
Unexpected exception when trying to log SystemClock version

### ISSUE THREAD

**Issue by HaloFour** [open-telemetry/opentelemetry-java#2455]
**Describe the bug**
Noticed the following exception being logged when attempting to access `SystemClock`:

```
2021-01-07 10:18:15,287 restartedMain WARN org.apache.logging.log4j.spi.AbstractLogger caught java.lang.IllegalArgumentException logging MessageFormatMessage: Using the APIs optimized for: {} java.lang.IllegalArgumentException: can't parse argument number: 
    at java.base/java.text.MessageFormat.makeFormat(MessageFormat.java:1454)
    at java.base/java.text.MessageFormat.applyPattern(MessageFormat.java:492)
    at java.base/java.text.MessageFormat.<init>(MessageFormat.java:391)
    at org.apache.logging.log4j.message.MessageFormatMessage.formatMessage(MessageFormatMessage.java:115)
    at org.apache.logging.log4j.message.MessageFormatMessage.getFormattedMessage(MessageFormatMessage.java:87)
    at org.apache.logging.log4j.core.async.InternalAsyncUtil.makeMessageImmutable(InternalAsyncUtil.java:41)
    at org.apache.logging.log4j.core.impl.MutableLogEvent.setMessage(MutableLogEvent.java:217)
    at org.apache.logging.log4j.core.impl.ReusableLogEventFactory.createEvent(ReusableLogEventFactory.java:100)
    at org.apache.logging.log4j.core.config.LoggerConfig.log(LoggerConfig.java:408)
    at org.apache.logging.log4j.core.config.AwaitCompletionReliabilityStrategy.log(AwaitCompletionReliabilityStrategy.java:63)
    at org.apache.logging.log4j.core.Logger.logMessage(Logger.java:153)
    at org.apache.logging.log4j.spi.ExtendedLoggerWrapper.logMessage(ExtendedLoggerWrapper.java:222)
    at org.apache.logging.log4j.spi.AbstractLogger.log(AbstractLogger.java:2110)
    at org.apache.logging.log4j.spi.AbstractLogger.tryLogMessage(AbstractLogger.java:2198)
    at org.apache.logging.log4j.spi.AbstractLogger.logMessageTrackRecursion(AbstractLogger.java:2152)
    at org.apache.logging.log4j.spi.AbstractLogger.logMessageSafely(AbstractLogger.java:2135)
    at org.apache.logging.log4j.spi.AbstractLogger.logMessage(AbstractLogger.java:2028)
    at org.apache.logging.log4j.spi.AbstractLogger.logIfEnabled(AbstractLogger.java:1899)
    at org.apache.logging.log4j.spi.AbstractLogger.log(AbstractLogger.java:1778)
    at org.apache.logging.log4j.jul.ApiLogger.log(ApiLogger.java:127)
    at io.opentelemetry.sdk.internal.JavaVersionSpecific.<clinit>(JavaVersionSpecific.java:44)
    at io.opentelemetry.sdk.internal.SystemClock.now(SystemClock.java:30)
```

This is in a Spring Boot 2.4 WebFlux project using log4j2 running on Java 15.  The type of the `CURRENT` in the static initializer is `Java9VersionSpecific`.  There appear to be no other deleterious effects and the project continues executing fine.

**What version and what artifacts are you using?**
Artifacts: `opentelemetry-sdk` (e.g., `opentelemetry-api`, `opentelemetry-sdk`, which exporters, etc)
Version: `0.13.1`
How did you reference these artifacts?

```
                implementation 'io.opentelemetry:opentelemetry-api:0.13.1'
                implementation 'io.opentelemetry:opentelemetry-sdk:0.13.1'
```

**Environment**
Compiler:
```
openjdk version "15" 2020-09-15
OpenJDK Runtime Environment AdoptOpenJDK (build 15+36)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 15+36, mixed mode, sharing)
```
OS: MacOS Big Sur 11.1

**Comment by jkwatson** [comment]
The line in the otel code is:
```
      logger.log(Level.FINE, "Using the APIs optimized for: {}", CURRENT.name());
```

I'm guessing this might be an issue with java 15 and your versions of log4j and the jul adapter, more than anything otel specific. Can you try putting a similar log line directly in your code and see what happens?

**Comment by jkwatson** [comment]
Turned out this was a mis-use of the JUL logging syntax. Fixed in #2464.
